mitiq/zne/tests/test_zne.py

```diff
@@ -493,32 +495,34 @@ def test_execute_with_zne_with_supported_circuits(circuit_type):
     assert abs(unmitigated - expected) > abs(zne_value - expected)
 
 
-@pytest.mark.parametrize("circuit_type", SUPPORTED_PROGRAM_TYPES.keys())
-def test_layerwise_execute_with_zne_with_supported_circuits(circuit_type):
+def test_layerwise_folding_with_zne():
     # Define a circuit equivalent to the identity
     qreg = cirq.LineQubit.range(2)
-    cirq_circuit = cirq.Circuit(
+    circuit = cirq.Circuit(
         cirq.H.on_each(qreg),
         cirq.CNOT(*qreg),
         cirq.CNOT(*qreg),
         cirq.H.on_each(qreg),
     )
-    # Convert to one of the supported program types
-    circuit = convert_from_mitiq(cirq_circuit, circuit_type)
-    expected = generic_executor(circuit, noise_level=0.0)
-    unmitigated = generic_executor(circuit)
-    # Use odd scale factors for deterministic results
-    fac = RichardsonFactory([1, 3, 5])
-    # Layerwise-fold
+    circuit_depth = len(circuit)
+    mock_executor = Mock(side_effect=lambda _: random.random())
     layer_to_fold = 0
     fold_layer_func = get_layer_folding(layer_to_fold)
+    scale_factors = [1, 3, 5]
+    factory = RichardsonFactory(scale_factors)
 
-    zne_value = execute_with_zne(
-        circuit, generic_executor, factory=fac, scale_noise=fold_layer_func
+    execute_with_zne(
+        circuit, mock_executor, factory=factory, scale_noise=fold_layer_func
     )
-
-    # Test zero noise limit is better than unmitigated expectation value
-    assert abs(unmitigated - expected) > abs(zne_value - expected)
+    assert mock_executor.call_count == len(scale_factors)
```

Comment:
To answer your question from the description, I like what you are asserting on in here, and I agree with removing asserting on actual values. This will still test an e2e behavior, and it will not be flaky.